tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java

```diff
@@ -416,4 +416,13 @@ public void testXMLLiteralInterpolation() {
         Assert.assertEquals(returns[0].stringValue(),
                 "<elem>42|3.14|31.4444|this-is-a-string|<abc></abc></elem>");
     }
+
+    @Test(description = "Test interpolating xml when there are extra dollar signs")
+    public void testXMLLiteralWithExtraDollarSigns() {
+        BValue[] returns = BRunUtil.invoke(result, "testDollarSignOnXMLLiteralTemplate");
+        Assert.assertTrue(returns[0] instanceof BXML);
+        Assert.assertEquals(returns[0].stringValue(), "<foo id=\"hello $5\">hello</foo>");
+        Assert.assertEquals(returns[1].stringValue(), "<foo id=\"hello $$5\">$hello</foo>");
+        Assert.assertEquals(returns[2].stringValue(), "<foo id=\"hello $$ 5\">$$ hello</foo>");
+    }
```

Comment:
I think this test case is not yet enabled in the testng.xml.

Better to add this test case to `XMLTest` class in jballerina-unit-tests

Reply:
Missed this, thank you. Updated XMLTest.java too.